playground/index.html

```diff
@@ -0,0 +1,127 @@
+        let isVIP = false;
+        tokens.map(ticket => { if (ticket.token.ticketClass == "VIP") isVIP = true });
+        if (isVIP) document.getElementById("vip-only-section").style.display = "block";
+        tokens.map(
+          ticketToken => {
+            const container = document.createElement("li");
+            const tokenId = document.createTextNode(ticketToken.token.ticketId);
+            container.appendChild(tokenId);
+            document.getElementById("tokenCards").appendChild(container);
+          }
+        );
+      }, function (error) {
+        console.log("error", error);
+      });
+    }
+    // to handle the vip click event
+    function vipClickEvent() {
+      // 1. Get most up to date Tokens (Confirm if this should be done)
+      negotiatorAuthenticator.init().then(function (tokens) {
+        // 2. Gather all VIP tickets
+        const vipTickets = tokens.filter(ticket => (ticket.token.ticketClass == "VIP"));
+        // 3. Build a Html template of vip tickets
+        let ticketsToHtml = "";
+        vipTickets.map(function (ticket) {
+          ticketsToHtml += `
+          <li>
+            <button style="margin: 20px" data-id="${ticket.token.ticketId}" onClick="selectVipTicket(${ticket.token.ticketId})">
+              ${ticket.token.ticketId}
+            </button>
+          </li>
+        `;
+        });
+        // 4. Render the tickets inside the vip-tickets-list
+        document.getElementById("vip-tickets-list").innerHTML = ticketsToHtml;
+        // 5. Show Modal
+        document.getElementById("vip-container").style.display = 'block';
+      }, function (error) {
+        console.log("error", error);
+      });
+    }
+    // to handle the vip selection click event from the end user
+    function selectVipTicket(ticketId) {
+      event.stopPropagation();
+      // 1. Get most up to date Tokens (Confirm if this should be done each time)
+      negotiatorAuthenticator.init().then(function (tokens) {
+        // 2. Get the ticket using ID
+        const chosenTicket = tokens.filter(ticket => (ticket.token.ticketId == ticketId))[0];
+        // 3. Authenticator (non-Disney mode) - no owner - open modal to retrieve owner
+        if (chosenTicket.ownerAddress == null) {
+          // lead to email code modal process, created by Authenticator.
+          negotiatorAuthenticator.assetOwnerAddress(chosenTicket);
+        }
+        // 4. If owner address is found, authenticateAddress
+        if (chosenTicket.ownerAddress == currentUser.ownerAddress) {
+          // this will lead to sign-message, even if user typed the code in email.
+
+          // TODO: Don’t encapsulate 'authenticateAddress' - use web3 challenge
+          let web3 = new Web3('HTTP://127.0.0.1:7545');
+          // Gnache / Mock Private Key only
+          const signature = web3.eth.accounts.sign('Some data', '35335e2eac26772c5b1aed0114f7f635b99eb17fe8ef37cd68edd7acf093fa9c');
```

Comment:
Started to build logic to apply a web3 signature challenge before moving to next step. 

Questions:

1. How should the required control flow work at this point (e.g. should this hold a promise until resolved?)
2. What is the required user experience